qprofiler2/src/org/qcmg/qprofiler2/summarise/ReadIDSummary.java

```diff
@@ -1,241 +1,353 @@
+		for(int i = 0; i < columns.length; i ++) {
+			if(columns[i].size() > 0)
+				XmlUtils.outputTallyGroup( element,  (i+1)+"thColumnSplitByColon", columns[i] , false );
 		}
+				
+		XmlUtils.outputTallyGroup( element,  "Instrument", instruments , false );
+		XmlUtils.outputTallyGroup( element,  "Flow Cell Id", flowCellIds , false );
+		XmlUtils.outputTallyGroup( element,  "Run Id", runIds , false );
+		XmlUtils.outputTallyGroup( element,  "Flow Cell Lane", flowCellLanes , false );
+		outputTallyGroup(element,  "Tile Number", tileNumbers);
+		outputTallyGroup( element,  "Pair infomation", pairs);
+		outputTallyGroup( element,  "Index", indexes );
+		//merge two pool together
+		pool_random.addAll(pool_uniq);
 		
-		updateMap(flowCellLanes, params[1]);
-		updateMap(tileNumbers,  params[2] );
-		// skip x, and y coords for now..
-		getPairInfo(params[4]);
+		//output 20 qname randomly
+		element = XmlUtils.createMetricsNode(ele, "qnameExample", null);			
+		//incase pool_random size is 1 or 0 
+		for( int i = 0; i < Math.min(20,pool_random.size()-1) ; i ++ ) {
+			int pos = r.nextInt( pool_random.size()-1);
+			XmlUtils.outputValueNode(element, pool_random.get(pos), 1);
+		}	
+	}
+	
+	private void outputTallyGroup(Element parent, String name, Map<String, AtomicLong> tallys) {
+		Element e1 = XmlUtils.outputTallyGroup( parent, name, tallys , false );
+		if(e1 != null ) {
+			String v1 = (tallys.size() < tallySize )? tallys.size()+"" : tallySize + "+";
+			e1.setAttribute("no", v1);
+		}
 	}
-
-	private <T> void updateMap(ConcurrentMap<T, AtomicLong> map , T key) {
-		AtomicLong al = map.computeIfAbsent(key, k-> new AtomicLong());	
-		al.incrementAndGet();		 
-	}	
 
 	/**
-	 * 	//@ERR091788.3104 HSQ955_155:2:1101:13051:2071/2
-							//@ERR091788 - machine id
-							//3104 - read position in file
-							// HSQ955 - flowcell
-							// 155 - run_id
-							// 2 - flowcell lane
-							// 1101 - tile
-							// 13051 - x
-							// 2071 - y
-							// 2 - 2nd in pair
-	 * @throws Exception 
+	 * 
+	 * @param <T>
+	 * @param map
+	 * @param key
+	 * @return true if it is a new key added
 	 */
-	void parseFiveElementHeaderWithSpaces(String [] params) {
-		// split by space
-		String [] firstElementParams = params[0].split(" ");
-		if (firstElementParams.length != 2) {
-			throw new UnsupportedOperationException("Incorrect header format encountered in parseFiveElementHeader. Expected '@ERR091788.3104 HSQ955_155:2:1101:13051:2071/2' but recieved: " + Arrays.deepToString(params));
-		}
-		String [] machineAndReadPosition = firstElementParams[0].split("\\.");
-		if (machineAndReadPosition.length != 2) {
-			throw new UnsupportedOperationException("Incorrect header format encountered in parseFiveElementHeader. Expected '@ERR091788.3104 HSQ955_155:2:1101:13051:2071/2' but recieved: " + Arrays.deepToString(params));
+	private  boolean updateMap(ConcurrentMap<String, AtomicLong> map , String key) {
+		String key1 = key;	 	
+		if(!map.containsKey(key1)) { 
```

Comment:
You are hitting the map 3 times here - could this be done more efficiently?

Reply:
reduced to twice